internal/asm/f64/l2norminc_amd64.s

```diff
@@ -0,0 +1,110 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !noasm,!appengine,!safe
+
+#include "textflag.h"
+
+#define SUMSQ X0
+#define ABSX X1
+#define SCALE X2
+#define ZERO X3
+#define TMP X4
+#define ABSMASK X5
+#define INF X7
+#define INFMASK X11
+#define NANMASK X12
+#define IDX AX
+#define LEN SI
+#define INC BX
+#define X_ DI
+
+#define ABSMASK_DATA l2nrodata<>+0(SB)
+#define INF_DATA l2nrodata<>+8(SB)
+#define NAN_DATA l2nrodata<>+16(SB)
+// AbsMask
+DATA l2nrodata<>+0(SB)/8, $0x7FFFFFFFFFFFFFFF
+// Inf
+DATA l2nrodata<>+8(SB)/8, $0x7FF0000000000000
+// NaN
+DATA l2nrodata<>+16(SB)/8, $0xFFF8000000000000
+GLOBL l2nrodata<>+0(SB), RODATA, $24
+
+// func L2NormInc(x []float64, n, incX uintptr) (sum float64)
+TEXT ·L2NormInc(SB), NOSPLIT, $0
+	MOVQ n+24(FP), LEN
+	MOVQ incX+32(FP), INC
+	MOVQ x_base+0(FP), X_
+	CMPQ LEN, $0          // if len== 0 {return 0}
+	JZ   retZero
+
+	XORPS ZERO, ZERO
+	XORPS INFMASK, INFMASK
+	XORPS NANMASK, NANMASK
+	MOVSD $1.0, SUMSQ           // ssq = 1
+	XORPS SCALE, SCALE
+	MOVSD ABSMASK_DATA, ABSMASK
+	MOVSD INF_DATA, INF
+	SHLQ  $3, INC               // INC *= sizeof(float64)
+
+initZero:  // for ;x[i]==0; i++ {}
+	// Skip all leading zeros, to avoid divide by zero NaN
+	MOVSD   (X_), ABSX // absxi = x[i]
+	UCOMISD ABSX, ZERO
+	JP      retNaN     // if isNaN(x[i]) { return NaN }
+	JNZ     loop       // if x[i] != 0 { goto loop }
+	ADDQ    INC, X_    // i += inc
+	DECQ    LEN
+	JZ      retZero    // if --len == 0 { return 0 }
```

Comment:
Rather than using `--i == 0`, please put the pre-decrements as post decrements on their own lines.